web/src/pages/apd/proposed-budget/IncentivePayments.js

```diff
@@ -99,15 +151,30 @@ const IncentivePayments = ({
                       {isViewOnly ? (
                         <Dollars>{data.ehAmt[year][q] || ''}</Dollars>
                       ) : (
-                        <DollarField
-                          className="budget-table--input-holder"
-                          fieldClassName="budget-table--input__number"
-                          aria-labelledby={`q${q} eh-payments`}
-                          label={`ehAmt payments for ${year}, quarter ${q}`}
-                          labelClassName="sr-only"
-                          name={`ehAmt-payments-${year}-q${q}`}
-                          value={data.ehAmt[year][q] || ''}
-                          onChange={updateEHPayment(year, q)}
+                        <Controller
+                          name={`data.ehAmt.${year}.${q}`}
+                          control={control}
+                          render={({ field: { onChange, name, ...props } }) => (
+                            <DollarField
+                              {...props}
+                              name={name}
+                              className="budget-table--input-holder"
+                              fieldClassName={dollar_error(data.ehAmt[year][q]) ? 'budget-table--input__number-error' : 'budget-table--input__number'}
+                              aria-labelledby={`q${q} eh-payments`}
+                              label={`ehAmt payments for ${year}, quarter ${q}`}
+                              labelClassName="sr-only"
+                              value={data.ehAmt[year][q] || '0'}
+                              onChange={({ target: { value } }) => {
+                                setEHPayment(year, q, value);
+                                onChange(value);
+
+                                if (adminCheck) {
+                                  trigger();
+                                  console.log({errors})
```

Comment:
console.log here. also, does this if (adminCheck) need to be here since its already in the useEffect above?